src/app/shared/search/search-result.component.ts

```diff
@@ -89,12 +91,44 @@ export class SearchResultComponent implements OnChanges {
       this.resultView.columns = this.resultView.columns.sort((a: SearchResultViewColumn, b: SearchResultViewColumn) => {
         return a.order - b.order;
       });
+
+      this.hydrateResultView();
     }
     if (changes['page']) {
       this.selected.page = (changes['page']).currentValue;
     }
   }
 
+  /**
+   * Hydrates result view with case field definitions.
+   */
+  // A longer term resolution is to move this piece of logic to the backend
+  hydrateResultView(): void {
+    this.resultView.results.forEach(result => {
+      const caseFields = [];
+
+      Object.keys(result.case_fields).forEach(fieldId => {
```

Comment:
can we introduce a method 
`getCaseFieldsId()` on SearchResultViewItem and use it here: 
```
result.getCaseFieldsId().forEach(fieldId => ....